File: frontend/src/shared/utils/terminalCajero.ts
/** Terminal del cajero: sede en BD o inferida del username (ej. cajero.wendelyn.mga → Managua). */
export function resolverTerminalCajero(
  perfil: { sede?: string | null } | null | undefined,
  username?: string | null
): string | null {
  const sede = perfil?.sede?.trim();
  if (sede) return sede;

  const u = (username ?? '').toLowerCase();
  if (u.includes('.mga') || u.endsWith('mga')) return 'Managua';
  if (u.includes('.bfs') || u.includes('bluefields')) return 'Bluefields';
  if (u.includes('managua')) return 'Managua';
  if (u.includes('martinez') && u.includes('mga')) return 'Managua';
  if (u.includes('wendelyn') && !u.includes('mga')) return 'Bluefields';
  return null;
}

export function etiquetaTerminal(terminal: string | null): string {
  return terminal ?? 'Sin asignar';
}

File: mobile/lib/core/utils/terminal_cajero.dart
/// Terminal del cajero: sede en BD o inferida del username.
String? resolverTerminalCajero({String? sede, String? username}) {
  final s = sede?.trim();
  if (s != null && s.isNotEmpty) return s;

  final u = (username ?? '').toLowerCase();
  if (u.contains('.mga') || u.endsWith('mga')) return 'Managua';
  if (u.contains('.bfs') || u.contains('bluefields')) return 'Bluefields';
  if (u.contains('managua')) return 'Managua';
  if (u.contains('wendelyn') && !u.contains('mga')) return 'Bluefields';
  return null;
}

String etiquetaTerminal(String? terminal) => terminal ?? 'Sin asignar';
